.github/workflows/pre-build.yaml

```diff
@@ -0,0 +1,48 @@
+# this workflow will try to lint and build a node.js application
+#
+# this is useful for stages that you require to make sure everything is working
+# properly before creating a container image to be pushed on the cloud
+#
+name: pre_build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'main'
+      - 'ci/*' # branches that follows the pattern ci/* can access this workflow too
```

Comment:
Really solid idea! I always hate troubleshooting issues with commits to `main`, this is a great workaround.